charts/graph-node/templates/deployment.yaml

```diff
@@ -0,0 +1,123 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: graph-node
+  namespace: {{ .Values.global.namespaceOverride | default .Release.Namespace }}
+spec:
+  replicas: {{ .Values.graphNode.replicas }}
+  selector:
+    matchLabels:
+      app: graph-node
+  template:
+    metadata:
+      labels:
+        app: graph-node
+    spec:
+      containers:
+      - name: graph-node
+        image: {{ .Values.graphNode.image }}:{{ .Values.graphNode.tag }}
+        ports:
+        - containerPort: {{ .Values.graphNode.ports.http | default 8000 }}
+        - containerPort: {{ .Values.graphNode.ports.jsonRpc | default 8001 }}
+        - containerPort: {{ .Values.graphNode.ports.indexNode | default 8020 }}
+        - containerPort: {{ .Values.graphNode.ports.metrics | default 8030 }}
+        - containerPort: {{ .Values.graphNode.ports.subgraphMetrics | default 8040 }}
+        env:
+        - name: postgres_host
+          value: postgres
+        - name: postgres_user
+          value: {{ .Values.postgres.user }}
+        - name: postgres_pass
+          value: {{ .Values.postgres.password }}
+        - name: postgres_db
+          value: {{ .Values.postgres.database }}
+        - name: ipfs
+          value: ipfs:5001
+        - name: ethereum
+          value: {{ .Values.environment.ethereumNetwork }}:{{ .Values.environment.ethereumRPC }}
+        - name: GRAPH_LOG
+          value: info
```

Comment:
Prefer generally to pass env vars via config map to seperate configuration from deployment.